timeseries/src/autogluon/timeseries/models/statsmodels/abstract_statsmodels.py

```diff
@@ -0,0 +1,143 @@
+    when pickled.
+
+    ModelFitSummary is consumed by AbstractStatsmodelsModel._predict_using_fit_summary to generate predictions.
+    """
+
+    model_name: str
+    model_init_args: Dict[str, Any]
+    parameters: Dict[str, Any]
+
+
+class AbstractStatsmodelsModel(AbstractTimeSeriesModel):
+    statsmodels_allowed_init_args: List[str] = []
+    statsmodels_allowed_fit_args: List[str] = []
+    quantile_method_name: str = "pred_int"
+
+    def __init__(
+        self,
+        freq: Optional[str] = None,
+        prediction_length: int = 1,
+        path: Optional[str] = None,
+        name: Optional[str] = None,
+        eval_metric: str = None,
+        hyperparameters: Dict[str, Any] = None,
+        **kwargs,  # noqa
+    ):
+        name = name or re.sub(r"Model$", "", self.__class__.__name__)
+        super().__init__(
+            path=path,
+            freq=freq,
+            prediction_length=prediction_length,
+            name=name,
+            eval_metric=eval_metric,
+            hyperparameters=hyperparameters,
+            **kwargs,
+        )
+        self._fitted_models: Dict[str, ModelFitSummary] = {}
+
+    def _get_default_model_init_and_fit_args(self) -> Tuple[dict, dict]:
+        default_model_init_args = {}
+        default_model_fit_args = {}
+        unused_args = []
+        for key, value in self._get_model_params().items():
+            if key in self.statsmodels_allowed_init_args:
+                default_model_init_args[key] = value
+            elif key in self.statsmodels_allowed_fit_args:
+                default_model_fit_args[key] = value
+            else:
+                unused_args.append(key)
+        if len(unused_args) > 0:
+            logger.warning(
+                f" {self.name} ignores the following hyperparameters: {unused_args}. "
+                f"See the docstring of {self.name} for the list of supported hyperparameters."
+            )
+        return default_model_init_args, default_model_fit_args
+
+    def _fit_local_model(
+        self, timeseries: TimeSeriesDataFrame, model_init_args: dict, model_fit_args: dict
+    ) -> ModelFitSummary:
+        """Fit a single local model to the time series."""
+        raise NotImplementedError
+
+    def _fit(self, train_data: TimeSeriesDataFrame, time_limit: int = None, **kwargs):
+        train_hash = hash_ts_dataframe_items(train_data)
+        items_to_fit = [item_id for item_id, ts_hash in train_hash.iteritems() if ts_hash not in self._fitted_models]
+        default_model_init_args, default_model_fit_args = self._get_default_model_init_and_fit_args()
+        # TODO: Parallelize fitting
+        with statsmodels_warning_filter():
+            for item_id in items_to_fit:
+                self._fitted_models[train_hash.loc[item_id]] = self._fit_local_model(
+                    timeseries=train_data.loc[item_id],
+                    default_model_init_args=default_model_init_args,
```

Comment:
shouldn't these parameters be called `model_init_args` etc.?

Reply:
The logic here is that `default_model_init_args` and `default_model_fit_args` are obtained by splitting the `self._get_model_params()`. These 2 dicts contain the (hyper)parameters exposed to the user, and these might be different from the `model_init_args` that are passed to `statsmodels` inside the `_fit_local_model` function (e.g., we always expose a `seasonal_period` parameter to the user, but internally each model uses a different name).

Do you think this naming `default_model_init_args` and `model_init_args` is reasonable to distinguish the two sets of parameters (exposed vs. internal), or would something else make more sense?